tests/ui/test_ui_tools.py

```diff
@@ -1101,6 +1101,15 @@ def test_keypress_goback(self, key):
         self.help_view.keypress(size, key)
         assert self.controller.exit_popup.called
 
+    @pytest.mark.parametrize('key', [
+        'up', 'down', 'j', 'k', 'J', 'K',
+        'page up', 'page down', 'G', 'end',
+    ])
+    def test_keypress_navigation(self, key):
+        size = (200, 20)
+        self.controller.keypress(size, key)
+        assert self.controller.keypress.called
```

Comment:
As written, this test is always going to pass. `self.controller` is set to be a `Mock` in the first method, as the way these are currently set up (due to `autouse=True`) that method is a little like an `__init__` for each test call. So the `assert` is testing that an explicit function call you made on a mock was actually made...which it was, in the test itself :)

You likely want `self.help_view` instead of `self.controller` on line 1110. This is the object/function you're actually testing, and when reading tests you'll maybe see some kind of setup/mocking, always a function call of the thing that's being tested, and then the `assert` type checks which specify what the end result should be, given the setup and call.

You'd also want to re-think the form of the parametrize here and use  `keys_for_command` to generate lists of tuples, rather than hard-coding in all the keys in the parametrize - in particular since the `assert` code really needs to consider the 'input' key and the resulting key passed to urwid, which are not always the same.